### DISCUSSION TITLE
Change indexes name when changing the table name in a sqlalchemy.sql.schema.Table instance

### DISCUSSION THREAD

**Discussion by davidemiani** [sqlalchemy/sqlalchemy#8919] [Usage Questions]
Hello there.

Let'say we have a sqlalchemy.sql.schema.Table instance.
I have noticed that, when changing the name of the table, index names do not change accordingly with the name. To better understand what I am saying, take a look at the code below, with print results commented.

```
from copy import (
    deepcopy,
)
from sqlalchemy import (
    Column,
)
from sqlalchemy.orm import (
    declarative_base,
)
from sqlalchemy.sql.schema import (
    Table,
)
from sqlalchemy.dialects.postgresql import (
    BIGINT,
)


Base = declarative_base()


class Test(Base):
    __tablename__ = "test"
    __table_args__ = {"schema": "tmp"}

    id = Column(BIGINT, primary_key=True, index=True)


test_table: Table = Base.metadata.tables["tmp.test"]

print(f"Table name: {test_table.name}")  # test
print(f"Index names: {[index.name for index in test_table.indexes]}")  # ['ix_tmp_test_id']

test_table_2: Table = deepcopy(test_table)
test_table_2.name = "new_incredible_name"

print(f"Table name: {test_table_2.name}")  # new_incredible_name
print(f"Index names: {[index.name for index in test_table.indexes]}")  # ['ix_tmp_test_id']
```

Is there a way to have the index name automatically set to "ix_tmp_new_incredible_name_id"? If not, don't you think it could be a good idea to implement a setter for the Table.name property?

**Comment by CaselIT** [comment]
Hi,

copy of a table it's supported at all, so the behavior of `deepcopy(test_table)` is currently undefined.

Regarding the name of the indexes, in general they don't have any connections to the names of a table, so I don't think it would be wise to appropriate to connect them. SQLAlchemy does have a naming convention to create default names, but once they are created it's no longer taken into consideration.

**Reply by zzzeek** [reply]
you can copy a Table object using [Table.to_metadata](https://docs.sqlalchemy.org/en/14/core/metadata.html#sqlalchemy.schema.Table.to_metadata).  you'll note there's a "name" argument that allows you to give the copy a new name.

**Reply by CaselIT** [reply]
yes, but IIRC that does not change the name of any schema item connected to the table (like index, constraints, etc0

**Reply by zzzeek** [reply]
if it uses naming convention, it probably should.   not sure if it does right now.
